### ISSUE TITLE
How about add volume "/var/run/docker.sock:/var/run/docker.sock" to the runner container

### ISSUE THREAD

**Issue by chengweiv5** [drone/drone#1122]
Hi there, I'm looking at master HEAD and see that `docker` publish plugin does support to build & publish docker image for project.

Two required options are:
- docker_host
- image

From a end-user point of view, I think it's good if `docker_host` is optional because as a end-user, I'm not interested in where my image will to be built, should I?

As I think, the drone CI/CD service should provides all equipment I want, rather than bring my own docker host to it.

So to make user life easier, I think it's good if we make `docker_host` optional, by default, we'll use the default docker socket(`/var/run/docker.sock`)  to build user docker image.

To do that, drone must mounts `docker.sock` to runner container, in addition, is it doable to add another option to .drone.yml which let user specify what volumes mounted to runner container? Something like below:

```
docker:
    - /host/dir:/container/dir:RO
    - /host/dir2:/container/dir2
```

How do you think?

**Comment by bradrydzewski** [comment]
hey there, good news is that this is being addressed in the next release. We have a dedicated docker plugin. See https://github.com/drone-plugins/drone-docker which runs Docker-in-Docker

In addition, the new yaml format exposes much more docker functionality, including the ability to mount volumes. This means you could mount the Docker socket as a volume also. See https://github.com/drone/drone/issues/902

I recommend subscribing to issue #902. I'll post when we have a 0.4 alpha release available for testing.

**Comment by chengweiv5** [comment]
thanks @bradrydzewski, I'll check 0.4 branch later.

Where could I find drone roadmap? Or when drone 0.3 will be available, so as 0.4?
